docs/architecture/deep-dives/database-replicas.md

```diff
@@ -0,0 +1,41 @@
+# Read-Only Database Replicas
+
+## Context
+
+Bitwarden Utilizes
+[Azure SQL Hyperscale](https://learn.microsoft.com/en-us/azure/azure-sql/database/service-tier-hyperscale?view=azuresql)
+
+and its
+[high-availability replica](https://learn.microsoft.com/en-us/azure/azure-sql/database/service-tier-hyperscale-replicas?view=azuresql#high-availability-replica)
+available as a _read_ replica. This allows Bitwarden to double the max worker limit (along with
+compute, memory, and network throughput) without needing to scale up the primary database. The HA
+replica replicates the primary instance by mirroring the transaction log records.
+
+## Vision
+
+More read-only capabilities should be used in conjunction with read-only replicas, especially after
+stored procedures have been tuned and there are prominent performance bottlenecks relating to
+read-only queries.
+
+## Usage
+
+We currently use the read-only replica in two areas:
+
+1. Dapper repositories with the `ReadOnlyConnectionString`
+   [defined](https://github.com/search?q=repo%3Abitwarden%2Fserver%20path%3A%2F%5Esrc%5C%2FInfrastructure%5C.Dapper%5C%2FRepositories%5C%2F%2F%20ReadOnlyConnectionString&type=code).
```

Comment:
I'm not sure this link is needed. This is a required constructor parameter for the base Dapper repository class, so all repositories should have it defined. And Github's search isn't great, e.g. GroupRepository [has it defined](https://github.com/bitwarden/server/blob/main/src/Infrastructure.Dapper/AdminConsole/Repositories/GroupRepository.cs) but it doesn't show up in that search.